### ISSUE TITLE
Allow to specify which argument to interpolate inside `Tween.tween_method()`

### ISSUE THREAD

**Issue by MewPurPur** [godotengine/godot-proposals#6185]
### Describe the project you are working on

A 2D game that uses `tween_method()` for multiple purposes, mostly to interpolate shader params.

### Describe the problem or limitation you are having in your project

Many times, I want to use a `tween_property()` to interpolate a property, but that property can only be modified via setters that take the value as a 2nd or 3rd argument. Examples of this being `Gradient.set_point(point_idx, value)` and `ShaderMaterial.set_shader_parameter(param_name, value)`.

So I have to use a MethodTweener in those cases, however, those can only interpolate the first argument. Let's take the shader material example.

**What I wish were possible:** Provide the `set_shader_parameter` callable, bind to it the `param_name`, then have a way to specify that I want the second argument to be interpolated from `from` to `to`.

**What I have to do:** Because `tween_method()` can only interpolate the first argument, I must first make a separate function that takes a value as its first argument, as well as a shader material and a `param_name`. This function swaps the arguments around to pass them to `set_shader_parameter()`. Only now I can pass this function as a callable for `tween_method()` to do its thing.

This adds more lines of code and it breaks my flow as I might have to switch to another script that implements my custom function, instead of just using the ShaderMaterial API I'm familiar with.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Implement a way to specify which argument of the method passed to `tween_method()` gets interpolated. This would probably be best done via a new method in MethodTweener, say `on_argument()`, taking the argument's index and defaulting to 0. This way it would require no changes to existing code.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I don't know, probably add `int tweened_arg = 0` to MethodTweener, make it adjustable through the new method, and tweak the places that assume the 1st argument should be the one interpolated.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Can be worked around as explained above. But I believe it will be useful most of the time with `tween_method()`.

### Is there a reason why this should be core and not an add-on in the asset library?

I think it's essential for making `tween_method()` comfortable to use.

**Comment by KoBeWi** [comment]
So I assume you want something like this
```GDScript
create_tween().tween_method(set_shader_parameter, 0, 1, 1).on_argument(1)
```
But how to set the value of the argument you skipped?

I think #4920 would be sufficient solution. You could do
```GDScript
create_tween().tween_method(set_shader_parameter.prebind(&"param"), 0, 1, 1)
```

**Comment by MewPurPur** [comment]
Yeah, I tried to implement this and noticed the problem. I supposed it would be possible to bind starting from 0, while skipping the argument, but I couldn't get it to work :( And this makes it quite convoluted, too. #4920 should be better.

**Comment by hatmix** [comment]
Perhaps binding args to the left should be called `partial`.
